Editor/Editors/CinemachineFreeLookEditor.cs

```diff
@@ -95,27 +92,33 @@ public override void OnInspectorGUI()
             DrawExtensionsWidgetInInspector();
         }
 
-        string[] RigNames;
-        CinemachineVirtualCameraBase[] m_rigs;
-        UnityEditor.Editor[] m_editors;
-        void UpdateRigEditors()
+        static GUIContent[] s_RigNames = 
         {
-            RigNames = CinemachineFreeLook.RigNames;
-            if (m_rigs == null)
-                m_rigs = new CinemachineVirtualCameraBase[RigNames.Length];
-            if (m_editors == null)
-                m_editors = new UnityEditor.Editor[RigNames.Length];
-            for (int i = 0; i < RigNames.Length; ++i)
+            new GUIContent("Top Rig"), 
+            new GUIContent("Middle Rig"), 
+            new GUIContent("Bottom Rig")
+        };
+        int m_SelectedRig = 1;
+
+        UnityEditor.Editor m_rigEditor;
+        CinemachineVirtualCameraBase m_EditedRig = null;
+
+        void UpdateRigEditor()
+        {
+            CinemachineVirtualCamera rig = Target.GetRig(m_SelectedRig);
+            if (m_EditedRig != rig || m_rigEditor == null)
             {
-                CinemachineVirtualCamera rig = Target.GetRig(i);
-                if (rig == null || rig != m_rigs[i])
+                m_EditedRig = rig;
+                if (m_rigEditor != null)
+                {
+                    UnityEngine.Object.DestroyImmediate(m_rigEditor);
+                    m_rigEditor = null;
+                }
+                if (rig != null)
                 {
-                    m_rigs[i] = rig;
-                    if (m_editors[i] != null)
-                        UnityEngine.Object.DestroyImmediate(m_editors[i]);
-                    m_editors[i] = null;
-                    if (rig != null)
-                        CreateCachedEditor(rig, null, ref m_editors[i]);
+                    Undo.RecordObject(Target, "selected rig");
+                    Target.m_YAxis.Value = m_SelectedRig == 0 ? 1 : (m_SelectedRig == 1 ? 0.5f : 0);
```

Comment:
This puzzled me for a while, but now I understand what it does. When you select a rig, this activates the selected rig by moving the vcam rig on the y axis.